### PR TITLE
[Merged by Bors] - Feat: add lemmas about `Sum`

### PR DESCRIPTION
**Pull request by urkud**
This is the Lean 4 version of leanprover-community/mathlib#18184

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Sum/Basic.lean [hunk 1]
```diff
@@ -99,16 +99,26 @@ variable {x y : Sum α β}
...
 
-theorem getRight_eq_none_iff : x.getRight = none ↔ x.isLeft := by
+@[simp] theorem getRight_eq_none_iff : x.getRight = none ↔ x.isLeft := by
   cases x <;> simp only [getRight, isLeft, eq_self_iff_true]
 
 #align sum.get_right_eq_none_iff Sum.getRight_eq_none_iff
 
+@[simp] lemma getLeft_eq_some_iff {a : α} : x.getLeft = a ↔ x = inl a := by
+  cases x <;> simp only [getLeft, Option.some.injEq, inl.injEq]
+
```
Comment (kbuzzard) [inline Mathlib/Data/Sum/Basic.lean:114]:
```suggestion
```

File: Mathlib/Data/Sum/Basic.lean [hunk 2]
```diff
@@ -99,16 +99,26 @@ variable {x y : Sum α β}
...
+#align sum.get_left_eq_some_iff Sum.getLeft_eq_some_iff
+
+@[simp] lemma getRight_eq_some_iff {b : β} : x.getRight = b ↔ x = inr b := by
+  cases x <;> simp only [getRight, Option.some.injEq, inr.injEq]
+
```
Comment (kbuzzard) [inline Mathlib/Data/Sum/Basic.lean:119]:
```suggestion
```

**Review by kbuzzard — APPROVED** [review]
LGTM! I think the convention with #align nowadays is no blank line beforehand.

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**